.cicd/build.sh

```diff
@@ -4,7 +4,7 @@ set -eo pipefail
 . ./.cicd/helpers/general.sh
 mkdir -p "$BUILD_DIR"
 [[ -z "$DCMAKE_BUILD_TYPE" ]] && export DCMAKE_BUILD_TYPE='Release'
-CMAKE_EXTRAS="$CMAKE_EXTRAS -DCMAKE_BUILD_TYPE=\"$DCMAKE_BUILD_TYPE\" -DENABLE_MULTIVERSION_PROTOCOL_TEST=\"true\" -DAMQP_CONN_STR=\"amqp://guest:guest@localhost:5672\""
+CMAKE_EXTRAS="$CMAKE_EXTRAS -DCMAKE_CXX_FLAGS=\"-Werror\" -DCMAKE_BUILD_TYPE=\"$DCMAKE_BUILD_TYPE\" -DENABLE_MULTIVERSION_PROTOCOL_TEST=\"true\" -DAMQP_CONN_STR=\"amqp://guest:guest@localhost:5672\""
```

Comment:
We probably want Werror on both CXX_FLAGS and C_FLAGS

Reply:
It doesn't work out of the box, see https://buildkite.com/EOSIO/eosio-build-unpinned/builds/9927#af15773f-5a88-4c25-984a-63586ce212e6.

There are a few options here:
1. just don't use `CMAKE_C_FLAGS="-Werror"`
2. convert "webassembly/runtimes/eos-vm-oc/gs_seg_helpers.c" into c++ and use `extern "C"`
3. remove `target_compile_options( eos-vm INTERFACE "-Wno-register" )` in `eos-vm/CMakeLists.txt` and then use `set_source_files_properties` to set `-Wno-register` only on c++ files.

Reply:
maybe you can do something like
```
target_compile_options(eos-vm INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-Wno-register>)
```

It also doesn't seem all that bad to propagate `-Wno-register` to the root CMakeLists

Reply:
@spoonincode can you elaborate the goal here and why this is needed?
I am worried about the scenario “Local builds work and then it fails mysteriously on CI due to -Werror only on CI”

Reply:
The problem is every compiler and version of the compiler has different warnings. And warnings are moved between `Wall`, `Wextra`, and `Wpedantic` between different versions. It's hard to predict what compiler users are going to use, and there are lot more combinations in the wild than our meager CI suite. As an example, I could no longer build eosio on my Linux box because both gcc 11 & clang 13 had warnings. clang 13 failed with,
```
libraries/abieos/include/eosio/name.hpp:21:14: error: definition of implicit copy assignment operator for 'name' is deprecated because it has a user-declared copy constructor [-Wdeprecated-copy,-Werror]
   constexpr name(const name&) = default;
```
This _is_ an interesting warning, but it's not worth shooting down some poor user's build due to a benign warning like this.

> I am worried about the scenario “Local builds work and then it fails mysteriously on CI due to -Werror only on CI”

This is a problem regardless (well, except for the "mysteriously" part, as long as one bothers to actually read the output) -- local developers are probably going to only build on one compiler locally before pushing, not the handful that are in CI.

That said, personally I would have had with a cmake option (defaulted to off) that enabled Werror, so a developer can easily toggle and leave this on locally. I'm indifferent on that aspect though.